### PR TITLE
remove bourbon

### LINKED ISSUE #875: Remove bourbon dependency
follow-up of https://github.com/palantir/blueprint/pull/719#issuecomment-281855926, https://github.com/palantir/blueprint/issues/123

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

🔥 remove `bourbon` dependency `core` (and devDep from all other packages)

we used three things from bourbon:

1. `@mixin font-face()` for icons: added our own `font-face` mixin based on original Bourbon source
2. `@mixin position()` was used heavily as shorthand for position properties. I added `@mixin position-all(absolute, 0)` for the simple set-all-sides-to-the-same-value case, and expanded all other usages to the individual CSS properties.
3. `@mixin prefixer()` was used for `appearance` prop in a few places; replaced with actual CSS properties.

### CODE REVIEW

File: packages/core/src/common/_mixins.scss [hunk 1]
```diff
@@ -105,3 +112,13 @@ $pt-dark-intent-text-colors: (
 @function centered-text($height, $line-height: floor($pt-font-size * $pt-line-height)) {
   @return floor(($height - $line-height) / 2);
 }
+
+// Removes the unit from a Sass numeric value (if present).
+// strip-unit(12px) => 12
+@function strip-unit($number) {
+  @if type-of($number) == "number" and not unitless($number) {
```
Comment (cmslewis) [inline packages/core/src/common/_mixins.scss]: Just verifying: are these built-in SCSS helpers?

File: packages/core/src/components/popover/_common.scss [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2015 Palantir Technologies, Inc. All rights reserved.
 // Licensed under the terms of the LICENSE file distributed with this project.
 
-@import "~bourbon/app/assets/stylesheets/bourbon";
+
```
Comment (cmslewis) [inline packages/core/src/components/popover/_common.scss]: Probably don't need two newlines in a row here.

**Comment by cmslewis** [comment]
p.s. @giladgray you have a merge conflict with `docs/package.json`.

File: packages/core/src/common/_mixins.scss (hunk 1 above)
Comment (giladgray) [reply packages/core/src/common/_mixins.scss]: indeed they are

File: packages/core/src/components/popover/_common.scss (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/popover/_common.scss]: ah snap one slipped through!

**Comment by giladgray** [comment]
@cmslewis this won't affect bundle size at all because Bourbon only exists compile time: it's a bunch of mixins and functions, used to _generate_ CSS.

File: packages/core/src/_typography.scss [hunk 3]
```diff
@@ -190,10 +190,11 @@ Markup:
 <pre>
 %pt-select {
   @include pt-button();
-  @include prefixer(appearance, none, webkit moz);
   border-radius: $pt-border-radius;
   height: $pt-button-height;
   padding: 0 ($input-padding-horizontal * 3) 0 $input-padding-horizontal;
+  -moz-appearance: none;
```
Comment (llorca) [inline packages/core/src/_typography.scss]: ooc, none of our tooling does automatic prefixing by default?

File: packages/core/src/common/_mixins.scss [hunk 4]
```diff
@@ -105,3 +112,13 @@ $pt-dark-intent-text-colors: (
...
+    @return $number / ($number * 0 + 1);
```
Comment (llorca) [inline packages/core/src/common/_mixins.scss]: `$number / 1`?

File: packages/docs-app/src/styles/_colors.scss [hunk 5]
```diff
@@ -119,8 +119,8 @@ label.docs-color-scheme-label { margin: ($pt-grid-size * 2) 0; }
   }
 
   &::before {
-    @include position(absolute, -2px);
-    border-radius: 4px;
+    @include position-all(absolute, -$palette-border-radius / 2);
```
Comment (llorca) [inline packages/docs-app/src/styles/_colors.scss]: 👌

File: packages/core/src/common/_mixins.scss (hunk 4 above)
Comment (giladgray) [reply packages/core/src/common/_mixins.scss]: `12px / 1 == 12px`. but `12px / (0px + 1) == 12`
unit math, my friend!

File: packages/core/src/_typography.scss (hunk 3 above)
Comment (giladgray) [reply packages/core/src/_typography.scss]: oof actually no, we don't run autoprefixer on the generated CSS file. one of the things we lost in the (gulp) fire, along with comment stripping.

File: packages/core/src/common/utils/_font-face.scss [hunk 6]
```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the terms of the LICENSE file distributed with this project.
+
+// these implementations borrowed from Bourbon v5, and modified to fit our use cases.
+// changes: removed $asset-pipeline argument, added weight/style params like Bourbon v4.
+// https://github.com/thoughtbot/bourbon/blob/master/core/bourbon/library/_font-face.scss
+// https://github.com/thoughtbot/bourbon/blob/master/core/bourbon/utilities/_font-source-declaration.scss
+
+/// Generates an `@font-face` declaration. You can choose the specific file
```
Comment (adidahiya) [inline packages/core/src/common/utils/_font-face.scss]: why the triple slashes?

Comment (giladgray) [reply packages/core/src/common/utils/_font-face.scss]: I dunno, copy-pasted from bourbon source. I think they use it for doc comments? I'll revert to doubles

**Pull request merged**